libethereum/BlockChainSync.cpp

```diff
@@ -177,6 +177,15 @@ BlockChainSync::~BlockChainSync()
 	abortSync();
 }
 
+void BlockChainSync::onBlockImported(BlockHeader const& _info)
+{
+	//if a block has been added via mining or other block import function
+	//through RPC, then we should count it as a last imported block
+	RecursiveGuard l(x_sync);
+	m_lastImportedBlock = static_cast<unsigned>(_info.number());
```

Comment:
https://github.com/winsvega/cpp-ethereum/blob/5540e2520b32ff5efb75630e8dade2a85d1cb38d/libethereum/BlockChain.cpp#L867

I think this is the place where the latest best block is called. 
so it is not just import. but a successful block being added as a new top block.